web/tests/typeahead.test.js

```diff
@@ -242,3 +242,44 @@ run_test("sort_emojis: prioritise realm emojis", () => {
         emoji_list[0],
     ]);
 });
+
+run_test("upper_bound_lexicographically", () => {
```

Comment:
Cool. Other edge cases to test:
* an empty list of words
* an empty search word

Reply:
The way this function is used guaranteed there'd never be an empty word list or search word, which is why I'd skipped these cases earlier. But yes, for making it more future proof, I've added these in now. Thanks for the suggestion!